api/src/db/model/profile.ts

```diff
@@ -139,4 +144,26 @@ export default class ProfileModel extends Model {
       throw err;
     }
   }
+
+  async isNamespacePrefixUnique(prefix: string): Promise<boolean> {
+    const query = {
+      text: `
+        SELECT COUNT(*) FROM ${this.table}
```

Comment:
is the plan to fill in all the existing namespaces to db first?

Reply:
The prefix will be the licence plate we have now. This fn checks to make sure the newly minted one does not already exists in the db; its super unlikely that it would given its 7 random characters but its best to check to make sure.